### PR TITLE
Add the possibility to fork a game from replay

### PR DESCRIPTION
**Pull request by krystofNovotny**
Added a 'fork game' button to the replay mode. Clicking the button copies the game as your practice game from the turn which is currently shown in the replay.

### CODE REVIEW

File: src/game-app.js [hunk 1]
```diff
@@ -340,6 +340,10 @@ var buttons = {
     $('#game').toggleClass('is-busy');
     return false;
   },
+  fork: async () => {
+    let gameId = game.state._data.id;
```
Comment (Enijar) [inline src/game-app.js:344]: This should be `const` not `let`.

File: src/server/GameService.js [hunk 2]
```diff
@@ -272,6 +272,15 @@ class GameService extends Service {
 
     return game.id;
   }
+
+  async onForkGameRequest(client, gameId, turn) {
+    this.debug(`forkGame: gameId=${gameId}`);
+    let newGame = await dataAdapter.forkGame(gameId, this.clientPara.get(client.id).playerId);
```
Comment (Enijar) [inline src/server/GameService.js:278]: This should be `const` not `let.

File: src/data/FileAdapter.js [hunk 3]
```diff
@@ -107,6 +107,19 @@ export default class {
     return game;
   }
 
+  async forkGame(gameId, playerId) {
+    let game = await this.getGame(gameId);
+    let newId = uuid();
+    await this._lockAndCopyFile(`game_${game.id}`, `game_${newId}`);
+    let newGame = await this.getGame(newId);
```
Comment (Enijar) [inline src/data/FileAdapter.js:114]: Change `let` to `const` vars in this method.

File: src/game-app.js (hunk 1 above)
Comment (krystofNovotny) [reply src/game-app.js:344]: I'd rather remove this altogether, which I did

**Comment by Enijar** [comment]
@krystofNovotny fair point.

**Comment by pongstylin** [comment]
There are a number of issues with this pull request, but mostly philosophical.  The only practical issue is that it is one button too many.  On PC screens, the buttons overlap with the board.  But it is not so severe that I will hold back pushing this to the beta server.

**Pull request merged**